File: mechanical/src/main/scala/resonantinduction/mechanical/process/mixer/MixerNode.java
package resonantinduction.mechanical.process.mixer;

import net.minecraftforge.common.ForgeDirection;
import resonant.api.grid.INodeProvider;
import resonantinduction.core.interfaces.IMechanicalNode;
import resonantinduction.mechanical.energy.grid.MechanicalNode;

public class MixerNode extends MechanicalNode
{

    public MixerNode(INodeProvider parent)
    {
        super(parent);
    }
    
    @Override
    public boolean inverseRotation(ForgeDirection dir, IMechanicalNode with)
    {
        return dir == ForgeDirection.DOWN;
    }

}

File: src/main/scala/edx/mechanical/mech/process/mixer/MixerNode.scala
package edx.mechanical.mech.process.mixer

import edx.mechanical.mech.grid.NodeMechanical
import net.minecraftforge.common.util.ForgeDirection
import resonantengine.api.graph.INodeProvider

/**
 * Node designed just for the Mixer to use
 * @param parent - instance of TileMixer that will host this node, should never be null
 */
class MixerNode(parent: INodeProvider) extends NodeMechanical(parent)
{
  override def canConnect(direction: ForgeDirection): Boolean =
  {
    return direction == ForgeDirection.DOWN || direction == ForgeDirection.UP
  }

  /*
  override def inverseRotation(dir: ForgeDirection): Boolean =
  {
      return dir == ForgeDirection.DOWN
  }*/
}